src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -561,14 +561,31 @@ private SeparatedSyntaxList<SwitchExpressionArmSyntax> ParseSwitchExpressionArms
 
             while (this.CurrentToken.Kind != SyntaxKind.CloseBraceToken)
             {
+                // Help out in the case where a user is converting a switch statement to a switch expression. Note:
+                // `default(...)` and `default` will also be consumed as a legal syntactic patterns (though the latter
+                // will fail during binding).  So if the user has `default:` we will recover fine as we handle the
+                // errant colon below.
+                var errantCaseOrDefault = this.CurrentToken.Kind == SyntaxKind.CaseKeyword
```

Comment:
I was confused here, shouldn't this just be called `errantCase`? A default token never goes in here.

Reply:
Bleagh.   That's a bad name on my part. I'll fix.

Reply:
done.